_posts/2017-11-10-choosing-a-cni-provider.md

```diff
@@ -0,0 +1,315 @@
+---
+title: Choosing a CNI Provider
+description: Choosing which CNI provider to deploy in Kubernetes
+header: Choosing a CNI Provider for Kuberentes
+date:   2017-10-28 10:46:52 MDT
+categories: kuberentes cni
+---
+
+Container Network Interface(CNI), is a library, under the umbrella of the  Cloud
```

Comment:
I think it's an interface specification in the first place, which comes with a reference implementation in Go, and that consists of a library and a set of CLI utils.